marshal/marshal.go

```diff
@@ -315,62 +315,72 @@ func (m *Marshaler) ClaimPartition(topicName string, partID int) bool {
 // Heartbeat will send an update for other people to know that we're still alive and
 // still owning this partition. Returns an error if anything has gone wrong (at which
 // point we can no longer assert we have the lock).
-func (m *Marshaler) Heartbeat(topicName string, partID int, CurrentOffset int64) error {
+func (m *Marshaler) Heartbeat(topicName string, partID int, offset int64) error {
 	topic, err := m.getClaimedPartitionState(topicName, partID)
 	if err != nil {
 		return err
 	}
 
+	// Attempt to commit offset, this is best-effort and we don't care if it fails
+	// since the canonical storage is in the heartbeat
+	if err := m.CommitOffsets(topicName, partID, offset); err != nil {
+		log.Warningf("[%s:%d] failed to commit offset during heartbeat: %s",
```

Comment:
Do you avoid the possibility of a partial rewind (in the case where `CommitOffsets` returns an error) by always using the heartbeat offset (instead of the committed offset) whenever one exists, even if the heartbeat offset is smaller than the committed offset?

If so, would it make sense to also include that change in this pull request?